addons/website/tests/test_get_current_website.py

```diff
@@ -134,3 +135,24 @@ def test_02_signup_user_website_id(self):
 
         user = self.env['res.users'].create({'website_id': website.id, 'login': 'sad@mail.com', 'name': 'Hope Fully'})
         self.assertTrue(user.website_id == user.partner_id.website_id == website)
+
+    def test_recursive_current_website(self):
+        Rule = self.env['ir.rule']
+        Rule.create({
+            'name': 'Recursion Test',
+            'model_id': self.env.ref('website.model_website').id,
+            'domain_force': [(1, '=', 1)],
+            'groups': [],
+        })
+        user = self.env.ref('base.user_admin')
+        user = user.with_user(user.id)
+        user.invalidate_cache()
+        failed = False
+        with MockRequest(self.env, website=self.website):
+            try:
+                user._update_last_login()
```

Comment:
Why not a simple search? The test would be better not relying on any business code, but just a basic orm search should trigger the error, no?